Aqi - Added term 'local' instead of having to use zip codes

davidofferman opened the pull request:
Hello,
In this update to Aqi (Air Quality Index), I added a new option to provide the word "local" instead of a zip code. This could be very useful if you are traveling to a different city and don't know the local zip code.

**A few questions...**
I'm not sure if I validate the location property correctly (I'm not a Perl developer). Is it sufficient enough to check if the `$loc` and `$loc->{postal_code}` return `true`, and to assume a valid postal code is present? Is there a more preferred way?

In the test file, I'm not sure if the `DDG::Request->new` method can be used along side the method of testing strings by themselves. In all the test files I've seen, they either use one or the other.

I was considering having this Instant Answer trigger without any additional content for the more specific trigger words, such as "air quality" and "air quality index". These queries would then request the postal code from the location variable. I would leave out the trigger "aqi" since that could end up over triggering. What are your opinions on this type of trigger to be added also?

---
Aqi Instant Answer
https://duck.co/ia/view/aqi

daxtheduck commented:
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

moollaza commented on lib/DDG/Spice/Aqi.pm:
```diff
@@ -20,9 +20,18 @@ spice to => 'http://www.airnowapi.org/aq/observation/zipCode/current/?format=app
+    my $query = shift;
+
+    if($query eq 'local') {
+        return unless ($loc && $loc->{postal_code});
```
This is probably the most robust approach to verifying the `$loc` object is defined and has the data we need:

https://github.com/duckduckgo/zeroclickinfo-spice/blob/a64e52ea83e7153c1dc33eaf68b2f96c3905f457/lib/DDG/Spice/OpenNIC.pm#L63

I would suggest you also verify that the country is USA and otherwise bail out as this API only supports US zipcodes

davidofferman commented:
I updated the checks for the `$loc` variables along with making sure the country is US. I also added a few additional comments to hopefully make future changes a breeze. 

I'm good with this commit if everyone else agrees.

moollaza commented on lib/DDG/Spice/Aqi.pm:
```diff
@@ -20,9 +20,31 @@ spice to => 'http://www.airnowapi.org/aq/observation/zipCode/current/?format=app
-     return $query; # we know it exists, it's a number and it has 5 digits;
+    my $query = shift;
+
+    if($query eq 'local') {
```
@davidofferman now that I think about this a little more, I think it's fair to trigger this when there is no remainder as long as the user is in the US.

That way queries like "air quality" would show this, similar to how "weather" shows the local weather.

moollaza replied:
I would also add `current` as an option: Something like `if $query eq "" || query m/^(local|current)$/ {` perhaps?

moollaza commented:
@davidofferman this is now live on beta! https://beta.duckduckgo.com/?q=aqi%2090210

jagtalon commented:
@davidofferman this is perfect! :+1:
